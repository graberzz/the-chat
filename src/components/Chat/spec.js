import React from 'react';
import { shallow } from 'enzyme';
import { Chat } from './';

const testRooms = [
  {
    id: '1',
    name: 'roomone',
    creator: 'admin',
  },
  {
    id: '2',
    name: 'roomtwo',
    creator: 'admin',
  },
];

const testMsgs = [
  {
    id: '1',
    message: 'msg1',
    timestamp: 1,
    user: {
      email: '@mail',
      username: 'test',
      avatar: 'avatar'
    },
  },
  {
    id: '2',
    message: 'msg2',
    timestamp: 3,
    user: {
      email: '@mail',
      username: 'test',
      avatar: 'avatar'
    },
  },
];

const testMatch = {
  params: {
    id: '1'
  },
};

describe('<Chat />', () => {
  const mockGetRoomMessages = jest.fn();
  it('matches snapshot', () => {
    const chat = shallow(
      <Chat
        typers={[]}
        rooms={testRooms}
        match={testMatch}
        messages={testMsgs}
        getRoomMessages={mockGetRoomMessages}
      />);

    expect(mockGetRoomMessages).toHaveBeenCalled();
    expect(chat).toMatchSnapshot();
  });
});