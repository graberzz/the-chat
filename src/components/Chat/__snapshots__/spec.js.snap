// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> matches snapshot 1`] = `
<React.Fragment>
  <MessageList
    messages={
      Array [
        Object {
          "id": "1",
          "message": "msg1",
          "timestamp": 1,
          "user": Object {
            "avatar": "avatar",
            "email": "@mail",
            "username": "test",
          },
        },
        Object {
          "id": "2",
          "message": "msg2",
          "timestamp": 3,
          "user": Object {
            "avatar": "avatar",
            "email": "@mail",
            "username": "test",
          },
        },
      ]
    }
  />
  <styled.div>
    <Connect(MessageInput)
      api={
        Object {
          "addRoom": [Function],
          "getRoomMessages": [Function],
          "login": [Function],
          "logout": [Function],
          "onAuthChange": [Function],
          "sendMessage": [Function],
          "setTyping": [Function],
          "unsetTyping": [Function],
          "watchNewMessages": [Function],
          "watchRooms": [Function],
          "watchTypers": [Function],
        }
      }
      roomId="1"
    />
  </styled.div>
</React.Fragment>
`;
